### ISSUE TITLE
Blow-up of the loss during training

### ISSUE THREAD

**Issue by zijianwang2000** [stefanradev93/BayesFlow#23]
Hi Stefan,

I am currently experiencing that for certain models (especially when the size of the dataset is slightly larger, e.g. n_obs = 100), the loss sometimes blows up extremely (say, about once every 40 epochs) so that the network is not converged after 100 epochs. Below is an example of such a blow-up (I simply printed the loss after each iteration):

..., -7.752036, -7.820761, -7.7039595, -7.6541286, -7.866893, -7.6999016, -7.7101583, -7.876931, -7.926651, 718.84845, 3543009.5, -1.1094699, 55.888943, 97.26874, 86.27302, 88.00319, 72.44775, 55.576897, 55.679665, 42.166874, 38.97898, 26.977058, 21.449322, 13.875741, 9.592149, 4.081973, 0.62280387, 0.8802163, -0.20799232, -0.26961386, -0.02687946, -1.470809, -0.33887288, -0.95399594, -1.1490471, -1.5857422, -1.2776537, -1.7386447, -1.5166757, -2.1432261, -2.240626, -2.8441887, -2.161209, -2.8351824, -2.9727767, -2.9907823, -3.5804815, -3.6286428, -3.659181, -3.566587, -3.8588657, -3.8712082, -3.9153986, -4.028116, -3.8262885, ...

Do I possibly need to change something in the optimizer setting or is it the number/size of affine coupling blocks that needs to be chosen more carefully? And is the default optimizer setting in the current BayesFlow package the same as the ones used for the BayesFlow paper?

Thanks a lot in advance!

Best regards
Zijian

**Comment by stefanradev93** [comment]
Hi Zijian,

I suspect that either your simulator produces weird outputs at some parameter values (e.g., huge numbers) or the learning rate you are using is too high for the network. You can also try exponential decay scheduler for the learning rate, e.g.,:

```
lr_schedule = tf.keras.optimizers.schedules.ExponentialDecay(
    initial_learning_rate=0.0005,
    decay_steps=5000,
    decay_rate=0.9,
    staircase=True,
)
```
and pass it to the Trainer. Let me know if that helps.

**Comment by zijianwang2000** [comment]
Hi Stefan,

thank you very much for your help!

I tried out the exponential decay scheduler for a model where I experienced the described blow-up of the loss. The result is that the blow-up is less severe now, i.e. it occurs less frequently and during the "stable phase", the loss seems lower than the ones achieved with the previous optimizer setting.

Let me maybe try out some other models and return to you in a couple of days. Thanks again!

Best regards
Zijian
